lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -133,11 +134,20 @@ import qualified Database.Sqlite as Sqlite
 -- Sqlite connection set up
 
 enableForeignKeys :: Sqlite.Connection -> IO ()
-enableForeignKeys conn = stmt >>= void . Sqlite.step
-    where stmt = Sqlite.prepare conn "PRAGMA foreign_keys = ON;"
+enableForeignKeys conn = do
+    stmt <- Sqlite.prepare conn "PRAGMA foreign_keys = ON;"
+    _ <- Sqlite.step stmt
+    Sqlite.finalize stmt
```

Comment:
this is needed to be able to close and avoid `SQLite3 returned ErrorBusy while attempting to perform close: unable to close due to unfinalized statements or unfinished backups`